File: Practica 3/Pila/Pila.java
package Pila;

public class Pila {
    private long arreglo[];
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        arreglo = new long[n];
        top = -1;
    }

    public void push(long e) {
        if (isFull()) {
            System.out.println("Pila llena");
        } else {
            top++;
            arreglo[top] = e;
        }
    }

    public long pop() {
        if (isEmpty()) {
            System.out.println("Pila vacia");
            return -1;
        } else {
            long e = arreglo[top];
            top--;
            return e;
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("Pila vacía");
            return -1;
        }
        return arreglo[top];
    }

    public boolean isEmpty() {
        return (top == -1);
    }

    public boolean isFull() {
        return (n - 1 == top);
    }
}

File: Practica 3/Pila/Pila.py
class Pila:
    def _init_(self, n):
        self.__arreglo = [None] * n
        self.__top = -1
        self.__n = n

    def push(self, e):
        if self.isFull():
            print("Pila llena")
        else:
            self.__top += 1
            self._arreglo[self._top] = e

    def pop(self):
        if self.isEmpty():
            print("Pila vacía")
            return None
        else:
            e = self._arreglo[self._top]
            self.__top -= 1
            return e

    def peek(self):
        if self.isEmpty():
            print("Pila vacía")
            return None
        else:
            return self._arreglo[self._top]

    def isEmpty(self):
        return self.__top == -1

    def isFull(self):
        return self._top == self._n - 1